net

```diff
@@ -511,30 +590,48 @@ def read_config(reraise = False):
     IGNORED_SSIDS = set(re.compile(ign) for ign in IGN.get('ssids') or [])
     VPN = raw_config.pop('vpn', {})
 
+    aliases = {}
+
     for name, raw_conn in raw_config.items():
+        # An alias
+        if isinstance(raw_conn, str):
+            def repl(m):
+                cmd, = [_f for _f in m.groups() if _f]
+                return run_or_die(cmd).strip()
+            aliases[name] = re.sub(r'\$\((.*?)\)|`(.*?)`', repl, raw_conn)
+            continue
         conn = {}
         for key, val in (raw_conn or {}).items():
-            # Assume UTF-8 encoding
-            if isinstance(val, unicode):
-                val = val.encode('utf8')
+            # FIXME: the change from Python 2 to 3 may have introduced all sorts
+            # of encoding errors
+
+            # # Assume UTF-8 encoding
+            # if isinstance(val, str):
+            #     val = val.encode('utf8')
             if key == 'wpa':
                 val = parse_wpa_conf(val)
             if key not in KNOWN_KEYS:
                 die('Unknown key: %s' % key)
-            if isinstance(val, unicode):
-                # The yaml decoder will convert
-                # non-ascii strings into unicode
-                # This is not what we want. We want
-                # e.g. '\xc3' in the yaml file to become b'\xc3',
-                # not u'\xc3'.
-                # This is an ugly hack to fix it.
-                val = val.encode('iso-8859-1')
+            # if isinstance(val, str):
```

Comment:
dead code?

Reply:
In principle yes, but dead code that we want to fix and re-add some day